OSDOCS#10460: Document configuring OAuth for the hosted cluster by using the web console (#75514)
openshift/openshift-docs: modules/hcp-configuring-oauth-console.adoc

// Module included in the following assemblies:
//
// * hosted_control_planes/hcp-authentication-authorization.adoc

:_mod-docs-content-type: PROCEDURE
[id="hcp-configuring-oauth-console_{context}"]
= Configuring the OAuth server for a hosted cluster by using the web console

> You can configure the internal OAuth server for your hosted cluster by using the {product-title} web console. You can configure OAuth for the supported identity providers such as `oidc`, `htpasswd`, `keystone`, `ldap`, `basic-authentication`, `request-header`, `github`, `gitlab`, and `google`. Adding any identity provider in the OAuth configuration removes the default `kubeadmin` user provider.

Reviewer: "supported identity providers such as" implies that this is an incomplete list, but the length of the list (9 providers) makes me wonder if this is a complete list? 

If it is meant to be only a selection of options from a larger list, consider just picking a few to highlight.

If it is a complete list, consider something like:
```suggestion
You can configure the internal OAuth server for your hosted cluster by using the {product-title} web console. 

You can configure OAuth for the following supported identity providers:
* `oidc`
* `htpasswd`
* `keystone`
* `ldap`
* `basic-authentication`
* `request-header`
* `github`
* `gitlab`
* `google`

Adding any identity provider in the OAuth configuration removes the default `kubeadmin` user provider.
```

Edit: I see this is taken from the CLI version, so use your best judgement here (if you want to tackle this for both later, that's fine imo)

.Prerequisites

* Log in as a user with `cluster-admin` privileges.

> * You created your hosted cluster.

Reviewer: These can be in past tense or present tense, but both should be the same for parallel construction :)

.Procedure

. Navigate to *Home* -> *API Explorer*.

. Use the *Filter by kind* box to search for your `HostedCluster` resource.
+

> image::config-oauth-console-hcp.png[Filtering the hosted cluster]

Reviewer: [I don't think this screenshot is needed](https://redhat-documentation.github.io/supplementary-style-guide/#screenshots). I would recommend removing it because the part of the interface you are guiding them to is unambiguous and you have included very clear instructions about putting `HostedCluster` in the box that says "Filter by kind".

Author: Agreed

You can configure the internal OAuth server for your hosted cluster by using the {product-title} web console.

You can configure OAuth for the following supported identity providers:

* `oidc`
* `htpasswd`
* `keystone`
* `ldap`
* `basic-authentication`
* `request-header`
* `github`
* `gitlab`
* `google`

Adding any identity provider in the OAuth configuration removes the default `kubeadmin` user provider.

.Prerequisites

* You logged in as a user with `cluster-admin` privileges.
* You created your hosted cluster.

.Procedure

. Navigate to *Home* -> *API Explorer*.

. Use the *Filter by kind* box to search for your `HostedCluster` resource.

. Click the `HostedCluster` resource that you want to edit.

. Click the *Instances* tab.

. Click the Options menu {kebab} next to your hosted cluster name entry and click *Edit HostedCluster*.

. Add the OAuth configuration in the YAML file:
+
[source,yaml]
----
spec:
  configuration:
    oauth:
      identityProviders:
      - openID: <1>
          claims:
            email: <2>
              - <email_address>
            name: <3>
              - <display_name>
            preferredUsername:

>               - <preferred_username> <4>

Reviewer 2: Honestly this is not worth holding up merging over so I won't but just something that you can consider fixing in a separate PR if you'd like - I think based on the description of this callout and the way you use callouts 1-3, maybe this callout should go on the line above this one instead.

Author: @skopacz1 Good catch! Opened a PR to fix this in both  - CLI and web console sections.
https://github.com/openshift/openshift-docs/pull/75647
